### DISCUSSION TITLE
JSON log parser: parse all fields when using parameters

### DISCUSSION THREAD

**Discussion by gdesmott** [grafana/grafana#51710] [Dashboards, Panels & Visualizations]
From the [json log parser documentation](https://grafana.com/docs/loki/latest/logql/log_queries/#parser-expression):

> Using | json label="expression", another="expression" in your pipeline will extract only the specified json fields to labels.

I need to use an expression in order to access an array. But by doing so all the other fields are no longer parsed, forcing me to do something like: `| json span="_spans[0]", message="message", session="session", participant_id="participant_id", target="_target", endpoint="endpoint", user_id="user_id", user_name="user_name" | `

This is especially annoying as the list of fields is not constrained.

It would be nice to have a way to keep parsing all fields even with using custom expressions.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
